modules/product/src/main/java/com/opengamma/strata/product/ReferenceSecurity.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product;
+
+import java.time.LocalDate;
+import java.util.Optional;
+
+import com.google.common.collect.ImmutableSet;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.market.ReferenceData;
+import com.opengamma.strata.collect.Messages;
+
+/**
+ * A single security that can be traded.
+ * <p>
+ * A security is one of the building blocks of finance, representing a fungible instrument that can be traded.
+ * This is intended to cover instruments such as listed equities and futures.
+ * It is intended that Over-The-Counter (OTC) instruments, such as an interest rate swap,
+ * are embedded directly within the trade, rather than handled as one-off securities.
+ * <p>
+ * A security may be referred to using a {@link SecurityId}.
+ * The identifier can be looked up in {@link ReferenceData}.
```

Comment:
Should this be "The security can be looked up in `ReferenceData` using the `SecurityId`?